qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -107,20 +113,68 @@ def disable_account(self):
                                    'Please use IBMQ.disable_accounts() to '
                                    'disable the account.')
 
-        if self._credentials is not None:
-            self._credentials = None
-            self._providers = OrderedDict()
-
-        else:
+        if not self._credentials:
             raise IBMQAccountError('No account is in use for this session.')
 
+        self._credentials = None
+        self._providers = OrderedDict()
+
     def load_account(self):
         """Authenticate against IBM Q Experience from stored credentials.
 
         Returns:
             AccountProvider: the provider for the default open access project.
+
+        Raises:
+            IBMQAccountError: if an IBM Q Experience 1 account is already in
+                use, or no IBM Q Experience 2 accounts can be found.
         """
-        raise NotImplementedError
+        if self._credentials:
+            # For convention, emit a warning instead of raising.
+            warnings.warn('Credentials are already in use.')
```

Comment:
shouldn't you return after this instead of continuing to do all the work?

Reply:
Good catch! Yes, it was an attempt to preserve previous behavior - I tried to find a better place for it along with a message closer to your suggestion in b6071c2